concourse/pipeline.yml

```diff
@@ -5,117 +5,133 @@ resource_types:
-
 resources:
-  - name: govuk-coronavirus-vulnerable-people-form
+  - name: git-master
     type: git
     icon: github-circle
     source:
       uri: https://github.com/alphagov/govuk-coronavirus-vulnerable-people-form
       branch: master
 
-  - name: govuk-coronavirus-vulnerable-people-form-feature-tests
+  - name: git-master-gems-dockerfile
     type: git
     icon: github-circle
     source:
       uri: https://github.com/alphagov/govuk-coronavirus-vulnerable-people-form
       branch: master
       paths:
+        - concourse/Dockerfile
         - Gemfile*
-        - .ruby-version
-        - spec/
 
   - name: every-day
     type: time
     source:
       interval: 24h
 
-  - name: govuk-coronavirus-vulnerable-people-form-travis-build
-    type: travis
-    icon: sync
-    source:
-      repository: alphagov/govuk-coronavirus-vulnerable-people-form
-      travis-token: ((travis-api-token))
-      branch: master
-      pro: true
-
-  - name: feature-tests-image
+  - name: tests-image
     type: docker-image
     icon: docker
     source:
       repository: ((readonly_private_ecr_repo_url))
-      tag: govuk-coronavirus-vulnerable-people-form-feature-tests
+      tag: govuk-coronavirus-vulnerable-people-tests-image
 
 jobs:
   - name: update-pipeline
     plan:
-      - get: govuk-coronavirus-vulnerable-people-form
+      - get: git-master
         trigger: true
       - set_pipeline: govuk-corona-vulnerable-people-form
-        file: govuk-coronavirus-vulnerable-people-form/concourse/pipeline.yml
+        file: git-master/concourse/pipeline.yml
 
-  - name: build-feature-tests-image
+  - name: build-tests-image
     serial: true
+    build_logs_to_retain: 100
     plan:
       - get: every-day
         trigger: true
-      - get: govuk-coronavirus-vulnerable-people-form-feature-tests
+      - get: git-master-gems-dockerfile
         trigger: true
-      - put: feature-tests-image
+      - put: tests-image
         params:
-          build: govuk-coronavirus-vulnerable-people-form-feature-tests
-          dockerfile: govuk-coronavirus-vulnerable-people-form-feature-tests/spec/Dockerfile
+          build: git-master-gems-dockerfile
+          dockerfile: git-master-gems-dockerfile/concourse/Dockerfile
+
+  - name: run-quality-checks
+    serial: true
+    plan:
+      - get: tests-image
+        passed: [build-tests-image]
+        trigger: true
+      - get: git-master
+        trigger: true
+      - task: run-tests-task
+        image: tests-image
+        config:
+          inputs:
+            - name: git-master
+          platform: linux
+          run:
+            dir: git-master
+            path: bash
+            args:
+              - -c
+              - |
+                set -eu
+                wget https://s3.eu-central-1.amazonaws.com/dynamodb-local-frankfurt/dynamodb_local_latest.tar.gz
+                sha256sum -c <<< "d77f5371c21868dd489b53782619b7f2e418d4b1068947a28d80082052ca1d64 dynamodb_local_latest.tar.gz"
```

Comment:
So... this was my stupid idea. This is extremely fragile and will break the moment amazon update the "latest" dynamo version.

I really don't like the idea of downloading it without doing a checksum check though...

they do publish the checksums [alongside here ](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.html), I just can't for the life of me work out how to do an automated comparison that doesn't seem like terrible command-line-fu.

Any cli blackbelts around to help?

Reply:
Alternatively we stick this in some docs and it's a feature whenever it breaks because a Dev can check for anything naferious?

Reply:
Or we tell it to download something other than latest, like a specific version, which we then stick to until we need to upgrade? Would be a good idea to match the AWS version?

Reply:
Yeah, I'd much prefer your third suggestion.

Other places in `org:alphagov` (GitHub search) that check the sha256sum of a file do it the way you've done it here.

Reply:
that's a relief, then we weren't totally insane going down this route.
Spent ages reading "how-to" guides that were all like... check it manually...

Reply:
looks like they have all the versions: https://s3.eu-central-1.amazonaws.com/dynamodb-local-frankfurt/